modal/cls.py

```diff
@@ -0,0 +1,33 @@
+# Copyright Modal Labs 2022
+import inspect
+from typing import Dict, Union
+from modal_utils.async_utils import synchronize_apis
+from .functions import _PartialFunction, PartialFunction, AioPartialFunction, _FunctionHandle
+
+
+def make_remote_cls_constructors(
+    user_cls: type,
+    partial_functions: Dict[str, Union[PartialFunction, AioPartialFunction]],
+    function_handles: Dict[str, _FunctionHandle],
+):
+    original_sig = inspect.signature(user_cls.__init__)  # type: ignore
+    new_parameters = [param for name, param in original_sig.parameters.items() if name != "self"]
```

Comment:
technically speaking, `self` isn't a keyword in Python, so this would break if someone creates a constructor like 

```python
def __init__(xyz, self: int = 2):
    ...
```

but like, lol

Reply:
Yeah I suppose I could check it's also the first argument 🤷